website/pages/docs/agent/options.mdx

```diff
@@ -2056,9 +2056,87 @@ Valid time units are 'ns', 'us' (or 'µs'), 'ms', 's', 'm', 'h'."
+    named `prometheus` that can be enabled explicitly here. It also requires the
+    `metrics_proxy` to be configured below and direct queries to a Prometheus
+    instance that has Envoy metrics for all services in the datacenter.
+
+  - `metrics_provider_files` ((#ui_config_metrics_provider_files)) - An optional array
+    of absolute paths to javascript files on the Agent's disk which will be
+    served as part of the UI. These files should contain metrics provider
+    implementations and registration enabling UI metric queries to be customized
+    or implemented for an alternative time-series backend.
+
+    ~> **Security Note:**  These javascript files are included in the UI with no
+    further validation or sand-boxing. By configuring them here the operator is
+    fully trusting anyone able to write to them as well as the original authors
+    not to include malicious code in the UI being served.
+
+  - `metrics_provider_options_json` ((#ui_config_metrics_provider_options_json)) -
+      This is an optional raw JSON object as a string which is passed to the
+      provider implementation's `init` method at startup to allow arbitrary
+      configuration to be passed through.
+
+  - `metrics_proxy` ((#ui_config_metrics_proxy)) - This object configures an
+    internal agent API endpoint that will proxy GET requests to a metrics
+    backend to allow querying metrics data in the UI. It simplifies deployment
+    where the metrics backend is not exposed externally to UI user's browsers.
+    It may also be used to augment requests with API credentials to allow
+    serving graphs to UI users without them needing individual access tokens for
+    the metrics backend.
+
+    The following sub-keys are available:
+
+    - `base_url` ((#ui_config_metrics_provider_base_url)) - This is required to
+      enable the proxy. It should be set to the base URL that the Consul agent
+      should proxy requests for metrics too. For example a value of
+      `http://prometheus-server` would target a Prometheus instance with local
+      DNS name "prometheus-server" on port 80. This may include a path prefix
+      which will then not be necessary in provider requests to the backend and
+      the proxy will prevent any access to paths without that prefix on the
+      backend.
+
+    - `add_headers` ((#ui_config_metrics_proxy_add_headers)) - This is an
+      optional list if headers to add to requests that are proxied to the
+      metrics backend. It may be used to inject Authorization tokens within the
+      agent without exposing those to UI users.
+
+      Each item in the list is an object with the following keys:
+
+      - `name` ((#ui_config_metrics_proxy_add_headers_name)) - Specifies the
+        HTTP header name to inject into proxied requests.
+
+      - `value` ((#ui_config_metrics_proxy_add_headers_value)) - Specifies the
+        value in inject into proxied requests.
+
```

Comment:
Would it be possible to show an example `ui_config` with metrics_proxy configured? (we show examples occasionally on this page, this one seems worth it).

Reply:
Should this feature have its own page describing it in general, how to enable it for prometheus, which envoy settings you need enabled to emit the right metrics, etc? Possibly also how it is used?

Reply:
Yeah I was thinking of adding more elaborate docs later when I actually finalize the provider interface! We could then link there. I'm not against an example here - I included an example for base_url just to make it clearer but I think the entire config and expectations of the backend and provider etc need their own page somewhere which I have down to do later for beta2 once I've made the changes to the provider.

Reply:
Hmm I went to add an example anyway but it's kinda confusing right now - the only supported provider (without going into detail about how to write your own which I don't want to yet) is Prometheus, but there are not really any plausible headers you need to send Prometheus since it has no Auth of it's own. I could make something up but not sure if that's super helpful.

So I'm going with this as bare-minimum docs for now with the understanding that before GA we'll have a more complete docs page with details about alternative providers and how to use this whole feature. Does that sound reasonable @preetapan?

Reply:
ok makes sense for now, given what you said above